typescript/web/src/pages/[workspaceSlug]/datasets/[datasetSlug]/index.tsx

```diff
@@ -30,6 +31,16 @@ const Body = () => {
     skip: typeof datasetSlug !== "string" || typeof workspaceSlug !== "string",
   });
 
+  if (
+    error &&
+    !isNil(error.networkError) &&
+    "result" in error.networkError &&
+    error.networkError.result.errors[0].message ===
+      "User not authorized to access dataset"
```

Comment:
As with #964 , please use `getApolloErrorMessage` so that the code is smaller and safer.